### PR TITLE
Mint adapter

### PR DESCRIPTION
**Pull request by alex-strizhakov**
- fixed request body streaming
- fixed SSL test
- added possibility for reusing connection, get body as stream/chunks

### CODE REVIEW

File: lib/tesla/adapter/mint.ex [hunk 1]
```diff
@@ -1,41 +1,47 @@
...
-      # set per module
-      defmodule MyClient do
-        use Tesla
-
-        adapter Tesla.Adapter.Mint
-      end
-
-      # set global custom cacert
-      config :tesla, Tesla.Adapter.Mint, cacert: ["path_to_cacert"]
+    Adapter for [mint](https://github.com/ericmj/mint)
```
Comment (chulkilee) [inline lib/tesla/adapter/mint.ex]: The repo is moved to https://github.com/elixir-mint/mint

Comment (alex-strizhakov) [reply lib/tesla/adapter/mint.ex]: 👍

**Comment by teamon** [comment]
@alex-strizhakov Please rebase on master, I did a small fix for 1.0.0 tests, maybe it will help.

**Comment by alex-strizhakov** [comment]
👌

**Pull request merged**